xcube/core/normalize.py

```diff
@@ -98,59 +103,138 @@ def normalize_dataset(ds: xr.Dataset,
     return ds
 
 
-def get_dataset_cube_subset(dataset: xr.Dataset,
-                            normalize: bool = False) \
-        -> Tuple[xr.Dataset, GridMapping]:
+def encode_cube(cube: xr.Dataset,
+                grid_mapping: Optional[GridMapping] = None,
+                non_cube_subset: Optional[xr.Dataset] = None) \
+        -> xr.Dataset:
     """
-    Get the data cube subset from *dataset* and the
-    cube's grid mapping.
+    Encode a *cube* with its *grid_mapping*, and additional variables in
+    *non_cube_subset* into a new dataset.
+
+    This is the inverse of the operation :func:decode_cube:
+
+        cube, gm, non_cube = decode_cube(dataset)
+        dataset = encode_cube(cube, gm, non_cube)
+
+    The returned data cube comprises all variables in *cube*,
+    whose dimensions should be ("time" , [...], y_dim_name, x_dim_name),
+    and where y_dim_name, x_dim_name are defined by *grid_mapping*, if
+    given.
+    If *grid_mapping* is not geographic, a new variable "crs" will
+    be added that holds CF-compliant attributes which encode the
+    cube's spatial CRS. *non_cube_subset*, if given may be used
+    to add other con-cube variables the to resulting dataset.
+
+    :param cube: data cube dataset, whose dimensions should
+        be ("time" , [...], y_dim_name, x_dim_name)
```

Comment:
That is simply not true.

Reply:
I looked it up yesterday, I found that it can only be used for persons. I looked it up again today in a more reliable source and found it can also be used for objects. So you're right, my bad.